// Copyright (c) 2016 Red Hat, Inc
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
// implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
option go_package = "git.openstack.org/openstack/oaktreemodel;oaktreemodel";

import "oaktreemodel/common.proto";

package oaktreemodel;

message Flavor {
  Location location = 1;
  string id = 2;
  string name = 3;
  bool is_public = 4;
  bool disabled = 5;
  uint32 ram = 6;
  uint32 vcpus = 7;
  uint64 disk = 8;
  uint32 ephemeral = 9;
  uint64 swap = 10;
  float rxtx_factor = 11;
  map<string, string> properties = 99;
}

message FlavorList {
  repeated Flavor flavors = 1;
}
